src/components/refactor/dialogs/two-windings-transformer-creation/ratio-tap-changer-pane/ratio-tap-changer-pane-utils.js

```diff
@@ -0,0 +1,205 @@
+    STEPS_REACTANCE,
+    STEPS_RESISTANCE,
+    STEPS_SUSCEPTANCE,
+    STEPS_TAP,
+    SUBSTATION_ID,
+    TAP_POSITION,
+    TARGET_DEADBAND,
+    TARGET_V,
+    TOPOLOGY_KIND,
+    TYPE,
+    VOLTAGE_LEVEL,
+} from 'components/refactor/utils/field-constants';
+import {
+    areArrayElementsOrdered,
+    areArrayElementsUnique,
+} from '../../../utils/utils';
+import yup from '../../../utils/yup-config';
+import {
+    getRegulatingTerminalEmptyFormData,
+    getRegulatingTerminalFormData,
+} from '../../regulating-terminal/regulating-terminal-form-utils';
+
+const ratioTapChangerValidationSchema = (id) => ({
+    [id]: yup.object().shape({
+        [ENABLED]: yup.bool().required(),
+        [LOAD_TAP_CHANGING_CAPABILITIES]: yup.bool().required(),
+        [REGULATING]: yup.bool().required(),
+        [TARGET_V]: yup
+            .number()
+            .nullable()
+            .min(0, 'TargetVoltageGreaterThanZero')
+            .when(REGULATING, {
+                is: true,
+                then: (schema) => schema.required(),
+            }),
+        [TARGET_DEADBAND]: yup
+            .number()
+            .nullable()
+            .min(0, 'TargetDeadbandGreaterThanZero'),
+        [LOW_TAP_POSITION]: yup
+            .number()
+            .nullable()
+            .max(100)
+            .when(ENABLED, {
+                is: true,
+                then: (schema) => schema.required(),
+            }),
+        [HIGH_TAP_POSITION]: yup
+            .number()
+            .nullable()
+            .min(yup.ref(LOW_TAP_POSITION), 'HighTapPositionError')
+            .max(100, 'HighTapPositionError')
+            .when(ENABLED, {
+                is: true,
+                then: (schema) => schema.required(),
+            }),
+        [TAP_POSITION]: yup
+            .number()
+            .nullable()
+            .when(ENABLED, {
+                is: true,
+                then: (schema) =>
+                    schema
+                        .required()
+                        .min(
+                            yup.ref(LOW_TAP_POSITION),
+                            'TapPositionBetweenLowAndHighTapPositionValue'
+                        )
+                        .max(
+                            yup.ref(HIGH_TAP_POSITION),
+                            'TapPositionBetweenLowAndHighTapPositionValue'
+                        ),
+            }),
+        [STEPS]: yup
+            .array()
+            .of(
+                yup.object().shape({
+                    [STEPS_TAP]: yup.number().required(),
+                    [STEPS_RESISTANCE]: yup.number(),
+                    [STEPS_REACTANCE]: yup.number(),
+                    [STEPS_CONDUCTANCE]: yup.number(),
+                    [STEPS_SUSCEPTANCE]: yup.number(),
+                    [STEPS_RATIO]: yup.number(),
+                })
+            )
+            .when(ENABLED, {
+                is: true,
+                then: (schema) => schema.min(1, 'GeneratePhaseTapRowsError'),
```

Comment:
I think this should be GenerateRatioTapRowsError

Reply:
fixed